admin_guide/cluster_management.rst

```diff
@@ -267,3 +267,174 @@ Output:
+      GROUP BY blocked_activity.query,
+               blocking_activity.query
+      ORDER BY 4
+    ) a
+  $cmd$);
+
+Example output:
+
+::
+
+  ┌───────────────────────────────────────────────────────────────────────────────────┐
+  │                               run_command_on_workers                              │
+  ├───────────────────────────────────────────────────────────────────────────────────┤
+  │ (localhost,5433,t,"")                                                             │
+  │ (localhost,5434,t,"{""update ads_102277 set name = 'new name' where id = 1; $ sel…│
+  │…ect * from ads_102277 where id = 1 for update; $ 1 $ 00:00:03.729519""}")         │
+  └───────────────────────────────────────────────────────────────────────────────────┘
+
+Querying the size of your shards
+--------------------------------
+
+This query will provide you with the size of every shard of a given distributed table, designated here with the placeholder :code:`my_distributed_table`:
+
+.. code-block:: postgresql
+
+  SELECT pg_size_pretty(result::bigint)
+  FROM run_command_on_shards('my_distributed_table', $cmd$
+    SELECT pg_table_size('%s');
+  $cmd$);
+
+Example output:
+
+::
+
+  ┌────────────────┐
+  │ pg_size_pretty │
+  ├────────────────┤
+  │ 8192 bytes     │
+  │ 125 MB         │
+  │ 8192 bytes     │
+  │ 8192 bytes     │
+  └────────────────┘
+
+For more info about distributed table size, see :ref:`table_size`.
+
+Identifying unused indices
+--------------------------
+
+This query will run across all worker nodes and identify any unused indexes for a given distributed table, designated here with the placeholder :code:`my_distributed_table`:
+
+.. code-block:: postgresql
+
+  SELECT *
+  FROM run_command_on_shards('my_distributed_table', $cmd$
+    SELECT array_agg(a) as infos
+    FROM (
+      SELECT (
+        schemaname || '.' || relname || '##' || indexrelname || '##'
+                   || pg_size_pretty(pg_relation_size(i.indexrelid))::text
+                   || '##' || idx_scan::text
+      ) AS a
+      FROM  pg_stat_user_indexes ui
+      JOIN  pg_index i
+      ON    ui.indexrelid = i.indexrelid
+      WHERE NOT indisunique
+      AND   idx_scan < 50
+      AND   pg_relation_size(relid) > 5 * 8192
+      AND   schemaname || '.' || relname = '%s'
+      ORDER BY
+        pg_relation_size(i.indexrelid) / NULLIF(idx_scan, 0) DESC nulls first,
+        pg_relation_size(i.indexrelid) DESC
+    ) sub
+  $cmd$);
+
+Example output:
+
+::
+
+  ┌─────────┬─────────┬───────────────────────────────────────────────────────────────────────┐
+  │ shardid │ success │                            result                                     │
+  ├─────────┼─────────┼───────────────────────────────────────────────────────────────────────┤
+  │  102008 │ t       │                                                                       │
+  │  102009 │ t       │ {"public.my_distributed_table_102009##stupid_index_102009##28 MB##0"} │
```

Comment:
Is there a reason this only shows unused on one shard? I'm not sure what would be actionable here, since the user can't selectively drop (without manual effort) an index on only one shard of a table. 

Also, what does the 28MB and the 0 represent at the end?

Reply:
Maybe it's actionable in that if the index is well utilized on most shards then keep it, but if used on few (or one) shards then you could consider dropping it.

Looks like `28MB##0` means the index takes up 28MB, and and has been used in an "index scan" zero times. I could modify the query with `json_build_object` so that the result columns have names rather than delimiting with `##`.

Reply:
> I could modify the query with json_build_object so that the result columns have names rather than delimiting with ##.

Could you do that? It might make the output more understandable.